diff --git a/include/hlsl++/platforms/scalar.h b/include/hlsl++/platforms/scalar.h
index f2db5ee..8eee6f6 100644
--- a/include/hlsl++/platforms/scalar.h
+++ b/include/hlsl++/platforms/scalar.h
@@ -550,7 +550,7 @@ HLSLPP_WARNING_POTENTIAL_DIVIDE_BY_0_END
 		return vector_int4(v1.x - v2.x * v3.x, v1.y - v2.y * v3.y, v1.z - v2.z * v3.z, v1.w - v2.w * v3.w);
 	}
 
-	hlslpp_inline vector_int4 _hlslpp_abs_epi32(const vector_int4& v) { return vector_int4(labs(v.x), labs(v.y), labs(v.z), labs(v.w)); }
+	hlslpp_inline vector_int4 _hlslpp_abs_epi32(const vector_int4& v) { return vector_int4(v.x >= 0 ? v.x : -v.x, v.y >= 0 ? v.y : -v.y, v.z >= 0 ? v.z : -v.z, v.w >= 0 ? v.w : -v.w); }
 
 	hlslpp_inline vector_int4 _hlslpp_cmpeq_epi32(const vector_int4& v1, const vector_int4& v2)
 	{
